### PR TITLE
Removed unused imports and variables; improved comments and messages …

### PR DESCRIPTION
**Pull request by pannarale**
Here is more low-level work on `pycbc_multi_inspiral`.
- Comments improved
- Messages to user improved (help for options and logging)
- Unused imports and variables removed

- [x] The author of this pull request confirms they will adhere to the [code of conduct](https://github.com/gwastro/pycbc/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: bin/pycbc_multi_inspiral [hunk 1]
```diff
@@ -68,25 +66,29 @@ parser.add_argument(
...
-    help="Cut triggers with NewSNR less than THRESHOLD",
+    help="Single IFO SNR threshold for trigger generation."
 )
+# Commenting out unused options: remove if they remain unused
+# parser.add_argument(
+#     "--newsnr-threshold",
+#     type=float,
+#     metavar='THRESHOLD',
+#     help="Cut triggers with NewSNR less than THRESHOLD",
+# )
```
Comment (pannarale) [inline bin/pycbc_multi_inspiral:80]: I will remove this in 2 PRs from now.  I have one coming up for further cleanup to this executable and then one on arguments that will affect these lines.

Comment (titodalcanton) [reply bin/pycbc_multi_inspiral:80]: Fair enough.

File: bin/pycbc_multi_inspiral [hunk 2]
```diff
@@ -228,82 +237,72 @@ args = parser.parse_args()
 init_logging(args.verbose)
 # Set GRB time variable for convenience
 t_gps = args.trigger_time
-# The following line aranges the ifos in alphabetical order so they are
-# always called in the same order.
+# Arange IFOs alphabetically so they are always called in the same order
```
Comment (titodalcanton) [inline bin/pycbc_multi_inspiral:240]:
```suggestion
# Arrange detectors alphabetically so they are always called in the same order
```

File: bin/pycbc_multi_inspiral [hunk 3]
```diff
@@ -228,82 +237,72 @@ args = parser.parse_args()
...
     args, args.instruments, inj_filter_rejector, dyn_range_fac=DYN_RANGE_FAC
 )
-# The following stores a dictionary of python slice objects, that indicate
-# where the segments begin and end,for each detector's timeseries.
+# Create a dictionary of Python slice objects that indicate where the segments
+# start and end for each detector timeseries.
 strain_segments_dict = strain.StrainSegments.from_cli_multi_ifos(
     args, strain_dict, args.instruments
 )
+# Python context
```
Comment (titodalcanton) [inline bin/pycbc_multi_inspiral:270]:
```suggestion
# Context manager to handle the various possible processing schemes
```

File: bin/pycbc_multi_inspiral [hunk 4]
```diff
@@ -382,11 +376,21 @@ with ctx:
...
-    # Clustering happens at the end of the template loop.
+    logging.info("Setting up MatchedFilterControl at each IFO")
+    # Prototype container for the output of MatchedFilterControl and
+    # waveform.FilterBank (see below).
+    # Use tlen of the first IFO as it is the same across IFOs.
+    template_mem = zeros(tlen, dtype=complex64)
+
+    # We are currently using the same matched-filter parameters for all
+    # IFOs. Therefore only one MatchedFilterControl setup is needed.
+    # This can be change if needed.
```
Comment (titodalcanton) [inline bin/pycbc_multi_inspiral:387]: I actually see one MatchedFilterControl for each detector, so is this comment still appropriate?

**Review by titodalcanton — APPROVED** [review]
Approved modulo a comment on a comment.

As a general note, despite the good use of comments, the algorithm is quite long and might benefit from being broken down into several functions.

File: bin/pycbc_multi_inspiral (hunk 4 above)
Comment (pannarale) [reply bin/pycbc_multi_inspiral:387]: Let me propose a different text.  What I meant is that (`segments[ifo]` aside) we call `MatchedFilterControl` the exact same way on all available detectors, whereas we would want a single IFO SNR threshold just for the two most sensitive ones.  This would require a non-zero `args.snr_threshold` for 2 detectors and a vanishing one for all other the others.

Comment (pannarale) [reply bin/pycbc_multi_inspiral:387]:
```
    # We are currently using the same matched-filter parameters for all
    # IFOs. Therefore we loop over one MatchedFilterControl setup.
    # This can be change if needed.
```

Comment (titodalcanton) [reply bin/pycbc_multi_inspiral:387]: Ah I see, then maybe rephrase "only one MatchedFilterControl setup is needed" as "all MatchedFilterControl instances are initialized in the same way"?

Comment (pannarale) [reply bin/pycbc_multi_inspiral:387]:
```
    # We are currently using the same matched-filter parameters for all
    # IFOs. Therefore only all MatchedFilterControl instances are initialized
    #  in the same way. This can be change if needed.
```
